coderedcms/models/page_models.py

```diff
@@ -1327,6 +1332,67 @@ def serve_submissions_list_view(self, request, *args, **kwargs):
         view = self.submissions_list_view_class.as_view()
         return view(request, form_page=self, *args, **kwargs)
 
+    def get_form(self, request, *args, **kwargs):
+        form_class = self.get_form_class()
+        form_params = self.get_form_parameters()
+        form_params.update(kwargs)
+
+        if request.method == 'POST':
+            return form_class(request.POST, request.FILES, *args, **form_params)
+        return form_class()
+
+    def contains_spam(self, request):
+        """
+        Checks to see if the spam honeypot was filled out.
+        """
+        if request.POST.get("cr-decoy-comments", None):
+            return True
+        return False
+
+    def process_spam_request(self, form, request):
+        """
+        Called when spam is found in the request.
+        """
+        messages.error(request, self.get_spam_message())
+        return self.process_form_get(form, request)
+
+    def get_spam_message(self):
+        return "Spam detected, form was not submitted."
```

Comment:
Should be translatable. I could see this message being taken the wrong way by a human (i.e. we are telling you that you are spam). Maybe update to read: "There was an error while processing your submission. Please try again."